ui/app/services/capabilities.ts

```diff
@@ -4,36 +4,82 @@
  */
 
 import Service, { service } from '@ember/service';
+import { assert } from '@ember/debug';
 
+import type AdapterError from '@ember-data/adapter/error';
+import type CapabilitiesModel from 'vault/vault/models/capabilities';
 import type StoreService from 'vault/services/store';
 
+interface Query {
+  paths?: string[];
+  path?: string;
+}
+
 export default class CapabilitiesService extends Service {
   @service declare readonly store: StoreService;
 
-  request = (apiPath: string) => {
-    return this.store.findRecord('capabilities', apiPath);
-  };
+  async request(query: Query) {
+    if (query?.paths) {
+      const { paths } = query;
+      return this.store.query('capabilities', { paths });
+    }
+    if (query?.path) {
+      const { path } = query;
+      const storeData = await this.store.peekRecord('capabilities', path);
+      return storeData ? storeData : this.store.findRecord('capabilities', path);
+    }
+    return assert('query object must contain "paths" or "path" key', false);
+  }
 
-  async fetchAll(apiPath: string) {
+  /*
+  this method returns a capabilities model for each path in the array of paths
+  */
+  async fetchMultiplePaths(paths: string[]): Promise<Array<CapabilitiesModel>> | AdapterError {
```

Comment:
nit: since we're just returning the promise (or throwing the error) we don't shouldn't `async` here or or `await` after the return

Reply:
yes! good catch, thank you. this changed a lot since the initial iteration.